moleculemaker/mmli-backend
# Feat/dg predictor new design

This pull requests:
- Returns reactants and products info for dg-predictor. 
   - in addition, add is_cofactor field for the molecule
- Refactor draw_chemical into a separate endpoint.

Test:
- Deployed in staging environment. 
- Go to https://mmli.fastapi.staging.mmli1.ncsa.illinois.edu/docs#/Files/get_results__bucket_name__results__job_id__get
- Enter job-type novostoic-dgpredictor
- Enter job-id 17465d9e770446eb9314ef385bac8aab
- Click Execute
- see results 
```
{
    "reaction": "C01745 + C00004 <=> N00001 + C00003 + C00001",
    "molecules": {
      "C01745": {
        "amount": -1,
        "name": "(E)-pinosylvin",
        "smiles": "Oc1cc(O)cc(/C=C/c2ccccc2)c1",
        "inchi": "InChI=1S/C14H12O2/c15-13-8-12(9-14(16)10-13)7-6-11-4-2-1-3-5-11/h1-10,15-16H/b7-6+",
        "inchi_key": "InChIKey=YCVPRTHEGLPYPB-VOTSOKGWSA-N",
        "metanetx_id": "MNXM4084",
        "kegg_id": "C01745",
        "is_cofactor": false,
        "structure": "...",
         ....
     }
    "novelMolecules": {
      "N00001": "c1ccc(cc1)/C=C/c1cccc(c1)O"
    },
    "gibbsEnergy": -121.74150463893119,
    "std": 3.513231952861522
}
```

or visit https://novostoic.frontend.staging.mmli1.ncsa.illinois.edu/thermodynamical-feasibility/result/17465d9e770446eb9314ef385bac8aab
or run a new dg-predictor to see result
